File: backend/src/models/Usuario.js
export class Usuario {
  constructor({ id, nome, email, data_nascimento, senha_hash }) {
    this.id = id;
    this.nome = nome;
    this.email = email;
    this.data_nascimento = data_nascimento;
    this.senha_hash = senha_hash;
  }
}

File: mobile/lib/models/usuario.dart
class Usuario {
  final String nome;
  final String email;
  final String dataNascimento;
  final String senha;

  Usuario({
    required this.nome,
    required this.email,
    required this.dataNascimento,
    required this.senha,
  });

  Map<String, dynamic> toJson() {
    return {
      'nome': nome,
      'email': email,
      'data_nascimento': dataNascimento,
      'senha': senha,
    };
  }
}

class UsuarioLogin {
  final String email;
  final String senha;

  UsuarioLogin({required this.email, required this.senha});

  Map<String, dynamic> toJson() {
    return {'email': email, 'senha': senha};
  }
}
